## Python Questions

In [1]:
# While loop with else clause

c = 1

while c<=5:
    print(f"count is {c}")
    c=c+1
else: 
    print("The loop completed normally without a break")

count is 1
count is 2
count is 3
count is 4
count is 5
The loop completed normally without a break


#### Q1) Number guessing game

In [2]:
n = 7
c = 1
_max = 3

while c<=_max:
    a = int(input(f"guess the number, guess number{c}"))
    c += 1

    if a==n:
        print("Number guessed right")
        break
    elif a>n:
        print("Correct number is less than guessed number")
    elif a<n:
        print("Correct number is greater than guessed number")
else:
    print(f"you exceeded the number of guessess limit, Correct number is {n} ")

guess the number, guess number1 4


Correct number is greater than guessed number


guess the number, guess number2 2


Correct number is greater than guessed number


guess the number, guess number3 1


Correct number is greater than guessed number
you exceeded the number of guessess limit, Correct number is 7 


#### Q2) Printing a Pattern

In [3]:
for i in range(0,5,1):
    for j in range(i):
        print("*", end="")
    print() #New line


*
**
***
****


In [4]:
# same pattern in reverse:

for i in range(4,0,-1):
    for j in range(i):
        print("*", end="")
    print()

****
***
**
*


#### Q3) Finding prime numbers

Number should be divisible by 1 and itself.

In [5]:
n = int(input("Enter a number"))
is_prime = True

if n<=1:
    is_prime = False
else:
    for i in range(2,n,1):
        if n%i == 0:
            is_prime = True
            break

if is_prime:
    print(f"Number {n} is prime")
else:
    print(f"Number {n} is not prime")         

Enter a number 5


Number 5 is prime


In [6]:
n = int(input("Enter a number"))
is_prime = True

if n<=1:
    is_prime = False
else:
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            is_prime = False
            break

if is_prime:
    print(f"Number {n} is prime")
else:
    print(f"Number {n} is not prime")

Enter a number 5


Number 5 is prime


Normal approach:

O(n)

Your optimized approach:

O(√n)

#### Q4) FizzBuzz: 
Write a program that prints numbers from 1 to 100. But for multiples of 3, print "Fizz" instead of the number, and for multiples of 5, print "Buzz". For numbers that are multiples of both 3 and 5, print "FizzBuzz".

In [7]:
for i in range(1,16,1):
    if (i % 3 ==0) and (i % 5 ==0):
        print("FizzBuzz")
    elif i % 3 ==0:
        print("Fizz")
    elif i % 5 ==0:
        print("Buzz")
    
    else:
        print(i)

1
2
Fizz
4
Buzz
Fizz
7
8
Fizz
Buzz
11
Fizz
13
14
FizzBuzz


#### Q5) Sum of Numbers: 
 Write a program that calculates the sum of all numbers from 1 to n, where n is a user-input.

In [8]:
n = int(input("enter a number"))
a = 0

for i in range(n+1):
    a = a+i

print(a)

enter a number 1


1


#### Q6) Factorial Calculator: 
Write a program that calculates the factorial of a number using a for loop. (The factorial of n is the product of all positive integers less than or equal to n).

In [9]:
n = int(input("enter a number"))
a = 1
for i in range(1,n+1):
    a = i*a
    print(a)

enter a number 6


1
2
6
24
120
720


#### Q7) Multiplication Table: 
Create a program that prints the multiplication table for a given number using a for loop.

In [10]:
n = int(input("enter a number"))

for i in range(1,11,1):
    print(f"{n}*{i}={n*i}")

enter a number 5


5*1=5
5*2=10
5*3=15
5*4=20
5*5=25
5*6=30
5*7=35
5*8=40
5*9=45
5*10=50


#### Q8) Fibonacci Sequence: 
Write a program to generate the first n numbers in the Fibonacci sequence, where n is provided by the user.

In [11]:
n = int(input("enter a number"))

a=0
b=1

for i in range(1,n+1,1):
    c = b
    b = a+b
    a = c
    if i == n:
        print(b)
    else: 
        print(b, end=",")

enter a number 6


1,2,3,5,8,13


#### Q9) Triangle Validator: 
Write a program that takes three inputs representing the lengths of the sides of a triangle and determines whether they can form a
valid triangle. (Hint: In a valid triangle, the sum of the lengths of any two sides is greater than the length of the remaining side.)

In [12]:
l1 = int(input("len  of 1st side of triangle"))
l2 = int(input("len  of 2nd side of triangle"))
l3 = int(input("len  of 3rd side of triangle"))

if (l1+l2) > l3 and (l2+l3) > l1 and (l3+l1) > l2:
    print("these can form a triangle")
else:
    print("these cannot form a triangle")

len  of 1st side of triangle 5
len  of 2nd side of triangle 5
len  of 3rd side of triangle 5


these can form a triangle


#### Concept - Nested list comprehension

In [13]:
matrix = [[1,2,3], [4,5,6], [7,8,9]]

flattened = [num for row in matrix for num in row]
flattened

[1, 2, 3, 4, 5, 6, 7, 8, 9]

#### Concept - Dictionary comprehension

In [14]:
numbers = [1,2,3,4,5]
squares_dict = {x: x**2 for x in numbers}
squares_dict

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25}

#### Concept - Creating Dictionary from 2 lists

In [15]:
fruits = ["apple", "banana", "cherry"]
prices = [1.2, 0.9, 2.5]

fruit_price = {fruit:price for fruit, price in zip(fruits, prices)}
fruit_price

{'apple': 1.2, 'banana': 0.9, 'cherry': 2.5}

#### Q10) List Manipulation: 
Write a function that takes a list of numbers and returns a new list containing only the even numbers from the original list, but
doubled.
Example:
    Input: [1, 2, 3, 4, 5]
    Output: [4, 8]
    # Only even numbers (2, 4) are kept and doubled

In [16]:
Input=[1, 2, 3, 4, 5]

output = [2*x for x in Input if x%2==0]
output

[4, 8]

#### Q11) Tuple Handling: 
Create a program that takes a list of tuples where each tuple contains a name and an age. Return a list of names for people who are a
least 18 years old.  

Example:  
Input: [("Alice", 20), ("Bob", 17), ("Charlie", 19), ("David", 16)]  
Output: ["Alice", "Charlie"]  
Only people 18 or older  

In [17]:
Input = [("Alice", 20), ("Bob", 17), ("Charlie", 19), ("David", 16)]

Output = [a for a,b in Input if b>=18]
Output

['Alice', 'Charlie']

#### Q12) Dictionary Operations: 

Given a dictionary of student names and their scores, write a program that returns a new dictionary containing only we students
who passed (score ›= 60) and their letter grades (A: 90-100, B: 80-89, C: 70-79, D: 60-69).  
Example:  
Input: {"Alice": 92, "Bob": 55, "Charlie": 78, "David": 85}  
Output: {"Alice": "A", "Charlie": "C", "David": "B"} # Bob is excluded for failing

In [18]:
Input = {"Alice": 92, "Bob": 55, "Charlie": 78, "David": 85}
Output = {}
for key,value in Input.items():
    if value>=90 and value<100:
        Output.update({key: "A"})
    elif value>=80 and value<90:
        Output.update({key: "B"})
    elif value>=70 and value<80:
        Output.update({key: "C"})
    elif value>=60 and value<70:
        Output.update({key: "D"})
    else:
        continue
Output

{'Alice': 'A', 'Charlie': 'C', 'David': 'B'}

#### Q13) Set Operations: 
Write a program that takes two lists and returns a list containing elements that appear in both lists, without duplicates ana in sorted
order.  
Example:  
Input: [1, 2, 2, 3, 4], [2, 3, 3, 5, 6]  
Output: [2, 3]  
Common elements, no duplicates, sorted  

In [19]:
listA = [4, 1, 2, 2, 3, 4]
listB = [4, 2, 3, 3, 5, 6]

output = {x for x in listA if x in listB}

o = list(output)
o.sort()
o

#print(o.sort()) - sorts list in-place

[2, 3, 4]

In [20]:
setA = set(listA)
setB = set(listB)

setA & setB

{2, 3, 4}

#### Q14) Comprehension Challenge: 
Given a list of strings, use a single comprehension to create a dictionary where the keys are the strings and the values are lengths of the strings, but only include strings with more than 3 characters.
Example:  
Input: ["a","abc","abcd","abcde"]  
Output: {"abcd": 4, "abcde": 5} # Only strings longer than 3 chars  

In [21]:
a = ["a","abc","abcd","abcde"]

d = {x:len(x) for x in a if len(x)>3}
d

{'abcd': 4, 'abcde': 5}

#### Q15). Advanced Iteration: 

Write a program that takes a list of dictionaries,   
where each dictionary represents a product with 'name', 'price', and 'category'   
Return the most expensive product in each category.   

Example:   
Input: [   
{"name": "Laptop", "price": 1080, "category": "Electronics"},   
{"name":"Headphones", "price": 100, "category": "Electronics"),   
{"name":"Shirt", "price": 20,"category": "Clothing"},   
{"name": "Jeans", "price": 50, "category": "Clothing"}   
]
Output: (   
"Electronics": ("name": "Laptop", "price": 1000, "category": "Electronics"),   
"Clothing": ("name": "Jeans", "price": 50,"category": "Clothing")   
}   

In [22]:
a = [   
{"name": "Laptop", "price": 1080, "category": "Electronics"},   
{"name":"Headphones", "price": 100, "category": "Electronics"},   
{"name":"Shirt", "price": 20,"category": "Clothing"},   
{"name": "Jeans", "price": 50, "category": "Clothing"}   
]

most_expensive_products = {}
for product in a:
    category = product["category"]
    price = product["price"]
    if (category not in most_expensive_products) or (price>most_expensive_products[category]['price']):
        most_expensive_products[category] = product
most_expensive_products    

{'Electronics': {'name': 'Laptop', 'price': 1080, 'category': 'Electronics'},
 'Clothing': {'name': 'Jeans', 'price': 50, 'category': 'Clothing'}}

#### Q16) Nested Data Structures: 
Given a list of dictionaries representing students, where each student has 'name',   
'grades' (a list), and 'courses' (a set) keys, find the student with the highest average grade.  
Example:  
Input: [  
{"name" :"Alice", "grades": [85, 90, 95], "courses": ("Math", "Science"}},  
{"name": "Bob", "grades": [70, 80, 90], "courses": ("History", "English", "Math"})}  
]  
Output: "Alice" # Average: 90 vs Bob's 80  

In [23]:
Input = [
{"name" :"Alice", "grades": [85, 90, 95], "courses": ("Math", "Science")},
{"name": "Bob", "grades": [70, 80, 90], "courses": ("History", "English", "Math")}
]

for student in Input:
    grades = student['grades']
    student['average_grades'] = sum(grades)/len(grades)
print(Input)   
    
highest_grade = 0

for x in Input:
    if x['average_grades']>highest_grade:
        highest_grade = x['average_grades']
        highest_grade_student = x['name']
highest_grade_student        

[{'name': 'Alice', 'grades': [85, 90, 95], 'courses': ('Math', 'Science'), 'average_grades': 90.0}, {'name': 'Bob', 'grades': [70, 80, 90], 'courses': ('History', 'English', 'Math'), 'average_grades': 80.0}]


'Alice'

#### Q17) Dictionary to List Conversion: 
Create a program that transforms a dictionary into a list of tuples, 
where each tuple contains the key, value, length of the string representation of the value.
    
Example:    
Input: ("a": 123, "b": "hello", "c": True)    
Output: [("a", 123, 3), ("b", "hello", 5), ("c","True", 4)]    


In [24]:
dictionary = {"a": 123, "b": "hello", "c": True}

_list = []

for key, value in dictionary.items():
    _list.append((key, value, len(str(value))))
_list

[('a', 123, 3), ('b', 'hello', 5), ('c', True, 4)]

#### Q18) Basic Function: 
Write a function called calculate_area that takes the radius of a circle as a parameter and returns its area (Tr * r°). Use the value
3.14159 for m.  
Example:  
Input: calculate_area(5)  
Output: 78.53975  

In [25]:
def calculate_area(r):
    pi = 3.145
    area = pi*r*r
    return area

calculate_area(5)

78.625

#### Q19) Default Parameters: 
Create a function called power that takes two parameters: base and exponent. Set the default value of exponent to 2. The
function should return the result of raising the base to the exponent.  
Example:  
Input: power (4)  
Output: 16 (4^2 = 16)  
Input: power (2, 3)  
Output: 8 (2^3 = 8)  

In [26]:
def power(base, exponent=2):
    result = base**exponent
    return result

power(2,3)

8

#### Q20) Multiple Return Values: 
Write a function called calculate_statistics that takes a list of numbers and returns three values: the mean (average),
median (middle value), and mode (most common value) of the list.
Example:  
Input: calculate_statistics([1, 2, 3, 2, 4, 2, 51)  
Output: (2.71, 2, 2)  

In [27]:
def calculate_statistics(_list):
    mean = round(sum(_list)/ len(_list),2)
    _list.sort()
    median = _list[len(_list)//2]
    dictionary = {num: _list.count(num) for num in _list}
    mode_number = max(dictionary.values())
    mode = [key for key, value in dictionary.items() if value == mode_number]
    return mean, median, mode

calculate_statistics([1, 2, 3,3,3, 2, 4, 2, 5])

(2.78, 3, [2, 3])

#### Q21) Function Composition: 
Write a function compose that takes two functions f and g as parameters and returns a new function that con.,.
a given input x.  
Example:  
def  add_one (x):  
  return x + 1    
def double(x):  
  return x * 2  
 
f = compose(add _one, double)  
f(3) -> 7 (add_one(double(3)) = add_one (6) = 7)  

In [28]:
def compose(f1, f2):
    def composed_function(x):
        return f1(f2(x))
    return composed_function
    
def add_one(x):
    return x+1

def double(x):
    return x+x

compose(add_one, double)(3)


7

#### Q22) Parameter Validation: 
Create a function called divide_safely that takes two parameters: numerator and denominator. The function should return the result of dividing the numerator by the denominator, but handle the case where the denominator is zero by returning a custom error message.

Example:  
Input: divide_safely(10, 2)  
Output: 5.0  
Input: divide_safely(10, 0)  
Output: "Error: Cannot divide by zero"  
32. Project 2 - Python Calculator - 1  

In [29]:
def divide_safely(numerator, denominator):
    try:
        division = numerator//denominator
        return division
    except ZeroDivisionError:
        print("Cannot divide by Zero")
    
divide_safely(10, 0) 

Cannot divide by Zero


#### Q23) Recursive Function: 
Write a recursive function called factorial that calculates the factorial of a given number. The factorial of n is the product of all
positive integers less than or equal to n (e.g., 5! = 5 × 4 × 3 × 2 × 1 = 120).
Example:  
Input: factorial(5)  
Output: 120  
Input: factorial(0)  
Output: 1  

In [30]:
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n-1)

factorial(1)

1

#### Project - Python Calculator

In [31]:
# calculator.py
def add(a, b):
    """Returns the sum of a and b."""
    return a+b

def subtract(a, b):
    """Returns the subtraction of a and b."""
    return a-b

def multiply(a, b):
    """Returns the product of a and b."""
    return a*b

def divide(a, b):
    """Returns the division of a and b; handle the division by zero."""
    if b == 0:
        return "Error: division by zero"
    return a/b

def display_history(history):
    """Display the history of calculations."""
    if not history:
        print("No calculations have been performed yet.")
    else:
        print("\nCalculations history:")
        for record in history:
            op, num1, num2, result = record
            print(f"{num1} {op} {num2} = {result}")


def main():
    # List to save history of calculations
    history = []

    # Menu of options to operations and symbol functions
    operations = {
        "1": ("+", add),
        "2": ("-", subtract),
        "3": ("*", multiply),
        "4": ("/", divide)
    }

    while True:
        print("\n----- Calculator Menu -----")
        print("1. Addition")
        print("2. Subtraction")
        print("3. Multiplication")
        print("4. Division")
        print("5. View Calculation History")
        print("6. Exit")

        choice = input("Enter your choice (1-6): ")

        if choice == "6":
            print("Exiting the calculator. Thank you!")
            break
        elif choice == "5":
            display_history(history)
            continue
        elif choice in operations:
            op_symbol, func = operations[choice]
            num1 = float(input("Enter the first number: "))
            num2 = float(input("Enter the second number: "))
            result = func(num1, num2)
            print("Result: ", result)
            history.append((op_symbol, num1, num2, result))
        else:
            print("Invalid choice. Please select one of the options.")

if __name__ == "__main__":
    main()



----- Calculator Menu -----
1. Addition
2. Subtraction
3. Multiplication
4. Division
5. View Calculation History
6. Exit


Enter your choice (1-6):  4
Enter the first number:  2
Enter the second number:  4


Result:  0.5

----- Calculator Menu -----
1. Addition
2. Subtraction
3. Multiplication
4. Division
5. View Calculation History
6. Exit


Enter your choice (1-6):  6


Exiting the calculator. Thank you!


#### Q24) File manipulation

In [32]:
import json
import os

contacts_file = 'contact.json'

def load_contacts():
    if os.path.exists(contacts_file):
        with open(contacts_file, "r") as file:
            try:
                return json.load(file)
            except json.JSONDecodeError:
                return "JSONDecodeError"
    else:
        return []

def save_contact(contacts):
    if False == (os.path.exists(contacts_file)):
        with open(contacts_file, "w") as file:
            json.dump({'Name': '', 'Ph.no': ''}, file, indent=4)
        
    with open(contacts_file, "r+") as file:
        json.dump(contacts, file, indent=4)
        

save_contact({'Name': 'Jithendra', 'Ph.no': 6300696935})        

#### Q25) Error Handling


1. Write a program that takes two numbers as input and divides them. Handle the case where the user enters a non-numeric value or tries to divide by zero.
2. Write a function that opens a file and reads its content. Handle cases where the file does not exist.
3. Create a function that asks the user to enter an age. Raise a ValueError if the age is negative or not a number.
4. Create a program that reads an integer from the user and ensures it is within the range 1-100. Raise an exception if it is out of range.

In [33]:
# Write a program that takes two numbers as input and divides them. 
#Handle the case where the user enters a non-numeric value or tries 
#to divide by zero.
try:
    numerator = int(input("Enter numerator"))
    denominator = int(input("Enter denominator"))
    print(numerator/denominator)
except ZeroDivisionError:
    print("Division not possible by zero")
except ValueError:
    print("Please enter valid number")


Enter numerator 5
Enter denominator 4


1.25


In [35]:
# Write a function that opens a file and reads its content.
#Handle cases where the file does not exist.
try:
    with open("test_file1.txt", "r") as file:
        print(file.read())
except FileNotFoundError:
    print(f'No such file name exists')


No such file name exists


In [36]:
# Create a function that asks the user to enter an age.
#Raise a ValueError if the age is negative or not a number.

try:
    age = int(input("Enter Age"))
    if age<0:
        print("Age cannot be less than Zero")
except ValueError:
    print(f"Please enter valid age, your entry is not a number")

Enter Age 0


In [37]:
# Create a program that reads an integer from the user and ensures 
#it is within the range 1-100. Raise an exception if it is out of range.

num = int(input("Enter integer"))

if not ((num<=100) and (num>=1)):
    print("Please enter a number in a range of 1-100")
else:
    print(f"you entered : {num}")

Enter integer 5


you entered : 5


#### OOP

1. Create a Person class with attributes name and age, and a method introduce() that prints "Hi, my name is (name) and 1 am (age) years old.".  
2. Modify the Car class to include a drive() method that prints "The (brand) is now driving.".  
3. Create a Rectangle class with attributes width and height. Add a method area that returns the area of the rectangle.  
4. Implement an Employee class with private attributes salary and methods set_salary) and get_salary) to modify and retrieve the salary.  
5. Create a Book class with attributes title, author, and year. Add a method get_info) that returns a formatted string describing the book.  

#### Q26) OOP
Create a Person class with attributes name and age, and a method introduce() that prints "Hi, my name is (name) and 1 am (age) years old.".


In [38]:
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age
    def introduce(self):
        return f"Hi, my name is {self.name} and I am {self.age} years old."

student1 = Person("jithendra", "25")
student1.introduce()

'Hi, my name is jithendra and I am 25 years old.'

#### Q27) OOP
Modify the Car class to include a drive() method that prints "The (brand) is now driving.".

In [39]:
class Car():
    def __init__(self, brand):
        self.brand = brand
    def drive(self):
        return f"The {self.brand} is now driving."

car1 =Car("Toyota")
car1.drive()
car1.brand

'Toyota'

#### Q28) OOP
Create a Rectangle class with attributes width and height. Add a method area that returns the area of the rectangle.

In [40]:
class Rectangle:
    def __init__(self, width, height):
        self.width = width
        self.height = height
    def area(self):
        a = self.width * self.height
        return a

area1 = Rectangle(20,20)
area1.area()

400

#### Q29) OOP
Implement an Employee class with private attributes salary and methods set_salary) and get_salary) to modify and retrieve the salary.


In [41]:
class Employee:
    def __init__(self, employee, salary):
        self.__salary = salary
        self.employee = employee
        
    def get_salary(self):
        return (self.employee, self.__salary)

    def set_salary(self, employee, salary):
        self.__salary = salary

employee1 = Employee("Ram", 123456)
employee1.get_salary()

('Ram', 123456)

In [42]:
employee1.set_salary("Ram", 78945612)

In [43]:
employee1.get_salary()

('Ram', 78945612)

In [44]:
employee1.__salary

AttributeError: 'Employee' object has no attribute '__salary'

#### Q30) OOP
Create a Book class with attributes title, author, and year. Add a method get_info) that returns a formatted string describing the book.

In [45]:
class Book:
    def __init__(self, title, author, year):
        self.title = title
        self.author = author
        self.year = year

    def get_info(self):
        print(f"{self.title} is written by {self.author} in {self.year}")

book1 = Book("'Do It Now'", "Leil Lowndess", "2025")
book1.get_info()

'Do It Now' is written by Leil Lowndess in 2025


#### Project Contact Manager

##### Contact Manager

This project is a contact manager that allows you to save, edit, and remove contacts from a JSON file. The program uses file manipulation, error handling (try-except blocks), and object-oriented programming.

##### Covered topics
- Object Oriented Programming
- Error handling
- File management

##### Features

- **List Contacts:** Displays all saved contacts.
- **Add Contact:** Allows you to enter a new contact (name, phone, email).
- **Edit Contact:** Updates an existing contact's details.
- **Remove Contact:** Deletes a contact from the JSON file.

In [23]:
class Contact_manager:
    directory = []
    import json
    with open("1.contacts.json", "r") as file:
        directory_data = json.load(file)
    for i in directory_data:
        directory.append({"Name":i['name'], "Key":i['phone']})
            
    def __init__(self):
        return

    @classmethod
    def Add_Contacts(cls, Name, Phone):
        cls.Name = Name
        cls.Phone_Number = Phone
        cls.directory.update({'Name':cls.Name, 'Phone_Number':cls.Phone_Number}) 
        return print(cls.directory)

    @staticmethod
    def List_Contacts():
        import json
        with open("1.contacts.json", "r") as file:
            directory_data = json.load(file)
            print(directory_data)

    def Edit_Contacts(self, Name, Phone):
        cls.Name = Name
        cls.Phone_Number = Phone
        cls.directory.update({'Name':cls.Name, 'Phone_Number':cls.Phone_Number}) 
        return print(cls.directory)

    @classmethod
    def Remove_Contacts(cls, Name, Phone):
        cls.Name_to_remove = Name
        cls.Phone_Number_to_remove = Phone
        if cls.Name_to_remove not in cls.directory:
            print("Name not found to be removed")
            return None
        else:
            cls.directory.remove({'Name':cls.Name_to_remove, 'Phone_Number':cls.Phone_Number_to_remove}) 
            return print(f"Contact {cls.Phone_Number_to_remove} is removed")
        

    
    
    def __str__(self):
        return 

Features = {"Option1":"List Contacts"
, "Option2": "Add Contact"
, "Option3": "Edit Contact"
, "Option4": "Remove Contact"
, "Option5": "Quit"
           }

for key, value in Features.items():
    print(f"{key}: {value}")
Feature_input = int(input("Choose your option"))
print(Feature_input)

if Feature_input == 5:
    pass
elif Feature_input == 1:
    Contact_manager.List_Contacts()
elif Feature_input == 2:
    Name = input("Enter the Name of person")
    Phone_number = int(input("Enter persons phone number"))
    person_details = Contact_manager.Add_Contacts(Name, Phone_number)
elif Feature_input == 3:
    print("Which details you want to edit 1.Name, 2.Phone Number, 3.Both")
    option_for_3 = int(input())
    if option_for_3 == 1:
        None

    elif option_for_3 == 2:
        None
elif Feature_input == 4:
    print("Enter the persons name to delete the contact")
    Name4 = input()
    Phone4 = int(input())
    person_remove_details = Contact_manager.Remove_Contacts(Name4, Phone4)
    
    
    


Option1: List Contacts
Option2: Add Contact
Option3: Edit Contact
Option4: Remove Contact
Option5: Quit


Choose your option 2


2


Enter the Name of person jith
Enter persons phone number 4444


AttributeError: 'list' object has no attribute 'update'